sunscreen_math/src/metal_impl/ristrettovec.rs

```diff
@@ -45,32 +45,51 @@ impl GpuRistrettoPointVec {
 
         let mut field_vec = Self { data, len };
 
-        let data_slice = unsafe { field_vec.buffer_slice_mut() };
+        // Turn the slice into a ptr into a usize so we can share it across threads.
+        // Each cell in the slice is going to be written to exactly 1 time, so a
+        // data race doesn't actually occur.
+        let data_ptr = unsafe { field_vec.buffer_slice_mut().as_ptr() } as usize;
 
-        for (i, p) in x.iter().enumerate() {
-            let x = p.0.X.to_u29();
-            let y = p.0.Y.to_u29();
-            let z = p.0.Z.to_u29();
-            let t = p.0.T.to_u29();
+        x.par_iter()
+            .map(|p| {
+                (
+                    p.0.X.to_u29(),
+                    p.0.Y.to_u29(),
+                    p.0.Z.to_u29(),
+                    p.0.T.to_u29(),
+                )
+            })
+            .enumerate()
+            .for_each(|(i, (x, y, z, t))| unsafe {
+                // Each cell in the
+                let data_ptr = data_ptr as *mut MaybeUninit<u32>;
 
-            let u29_len = x.len();
+                let u29_len = x.len();
 
-            for (j, w) in x.iter().enumerate() {
-                data_slice[(j + 0 * u29_len) * len + i].write(*w);
-            }
+                for (j, w) in x.iter().enumerate() {
+                    let elem = data_ptr.add((j + 0 * u29_len) * len + i);
```

Comment:
Huh, what are these `0 * u29_len` values? Looking at the types, it looks like `u29_len` is always `10: usize`, right? If so, why multiply it by zero?

Reply:
Oh, perhaps the next ones should still increment the offset like they used to? I.e. multiply by 0, 1, 2, 3 for x, y, z, t respectively?

Reply:
Exactly. The offsets aren't zero for y, z, t and this just makes the alignment be pretty.

Reply:
Gotcha. Just to be clear though, in the current diff, the offsets are currently `0` for all of `x, y, z, t`.

Reply:
Doh